designsafe/apps/rapid/openTopo_Data/scripts/centerpoint_datagenerator.py

```diff
@@ -103,6 +114,36 @@ def save_aggregated_results(output_filepath, all_features):
         geojson.dump(feature_collection, f, indent=2)
     logging.info(f"Aggregated results saved to: {output_filepath}")
 
+def fetch_otcatalog_api_response(file_path):
+    with open(file_path, 'r') as file1:
+        data = json.load(file1)
+    features = data['features']
+    # Specify the API endpoint URL
+    api_url="https://portal.opentopography.org/API/otCatalog"
+    # Specify default API query parameters
+    include_federated="false"
+    detail="true"
+    for feature in features:
+        center_point = feature['geometry']['coordinates']
+        minx, maxx, miny, maxy = calculate_bounding_box(center_point)
+        request_url=f'{api_url}?minx={minx}&miny={miny}&maxx={maxx}&maxy={maxy}&detail={detail}&include_federated={include_federated}'
+        response = requests.get(request_url)
```

Comment:
```suggestion
        response = requests.get(request_url)
        response.raise_for_status
```

Then you could drop the `if response.status_code == 200:` and the:
```
        else:
            logging.error(f"Failed to fetch data for {feature['properties']['id']}")
```

or if you want to keep the if and else and not use raise_for_status, then the `else` statement should be updated to throw an exception, so that an output file isn't written

```
        else:
            logging.error(f"Failed to fetch data for {feature['properties']['id']}")
            raise Exception("
```

Reply:
ah, is this the situation where there are some geometries in Data_Catalog_Spatial_Boundaries that don't correspond to datasets provided by `API/otCatalog`?

Reply:
Yes, currently the situation is happening for the arcticDEM datasets, but in case of multi-polygon data, there is always a possibility of the centroid not falling in any of the associated polygons. We are yet to figure out handling such exceptions.

Reply:
ah, i see.  hopefully this suggestion would work: https://github.com/DesignSafe-CI/portal/pull/1337/files#r1671274878 

we could also ask OT devs how best to make a single query to get everything.